### PR TITLE
Add TTL feature to allow metdata/userdata to expire after a set period of time

### PR DESCRIPTION
**Pull request by ScottGarman**
The default is one hour, but can be configured via a cmdline flag or environment variable. Set this parameter to zero to disable it.

### CODE REVIEW

File: pkg/api/v1/router_instance_metadata.go [hunk 1]
```diff
@@ -70,6 +70,14 @@ func (r *Router) instanceMetadataGet(c *gin.Context) {
 		return
 	}
 
+	// Treat metadata that's older then its TTL as non-existent
+	if metadata != nil && viper.GetDuration("cache_ttl") != 0 && metadata.UpdatedAt.Add(viper.GetDuration("cache_ttl")).Before(time.Now()) {
```
Comment (mikemrm) [inline pkg/api/v1/router_instance_metadata.go:74]: perhaps the ttl should be cached on the router so we're not having to re parse it for every request.

File: pkg/api/v1/router_instance_metadata_lookup_test.go [hunk 2]
```diff
@@ -70,4 +74,41 @@ func TestGetMetadataLookupByIP(t *testing.T) {
 			assert.Equal(t, testcase.expectedStatus, w.Code)
 		})
 	}
+
+	// Verify cache TTL settings are honored by setting the TTL to 1 second, sleeping to surpass that, and trying again
+	viper.Set("cache_ttl", time.Second)
+	time.Sleep(time.Second)
```
Comment (mikemrm) [inline pkg/api/v1/router_instance_metadata_lookup_test.go:80]:
```suggestion
	viper.Set("cache_ttl", -time.Second)
```

I believe this should work, and then we don't need to sleep.

File: pkg/api/v1/router_instance_metadata_test.go [hunk 3]
```diff
@@ -22,6 +24,8 @@ import (
 func TestGetMetadataByIP(t *testing.T) {
 	router := *testHTTPServer(t)
 
+	viper.Set("cache_ttl", 600*time.Second)
```
Comment (mikemrm) [inline pkg/api/v1/router_instance_metadata_test.go:27]: why is this being added here?

**Review by mikemrm — CHANGES_REQUESTED** [review]

File: pkg/api/v1/router_instance_metadata.go (hunk 1 above)
Comment (ScottGarman) [reply pkg/api/v1/router_instance_metadata.go]: I tried this, and I found that I could no longer change the value at runtime during the tests, even with some plumbing in the test mocks for the http server/router.

File: pkg/api/v1/router_instance_metadata_lookup_test.go (hunk 2 above)
Comment (ScottGarman) [reply pkg/api/v1/router_instance_metadata_lookup_test.go]: Cool idea. I've updated this and added a comment explaining why.

File: pkg/api/v1/router_instance_metadata_test.go (hunk 3 above)
Comment (ScottGarman) [reply pkg/api/v1/router_instance_metadata_test.go:27]: Yep, no need - removed now.

**Review by mikemrm — APPROVED** [review]

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**